### PR TITLE
feat : added filter functionality

### PR DESCRIPTION
**Pull request by rutvikpumak**
-> Added context and fetching videos & categories by API
-> Added Search functionality
-> Added video sort by category functionality
-> Added single video page redirects to particular that video

### CODE REVIEW

File: src/context/data/videoContext.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+const DataProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(videoReducer, initialState);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const {
+          data: { categories },
+        } = await axios.get("/api/categories");
```
Comment (SHIKHASINGH1506) [inline src/context/data/videoContext.js:22]: Here if your api call for category will fail, then videos api will also not work. You can use separate effect hooks for individual apis.

**Review by SHIKHASINGH1506 — APPROVED** [review]
Nothing major caught my eyes, just left one suggestion. 
Things that I like 
1. You've used de structuring very well
2. Hooks are self intuitive
Good to merge.

File: src/context/data/videoContext.js (hunk 1 above)
Comment (rutvikpumak) [reply src/context/data/videoContext.js:22]: Okay , will refactor it

**Review by xsaurabhsuthar — APPROVED** [review]
Everything looks clean and good.

**Pull request merged**